tictactoe/adapter-script-generation/src/main/kotlin/com/anaplan/engineering/azuki/tictactoe/adapter/scriptgen/TicTacToeScriptGenerationCheckFactory.kt

```diff
@@ -0,0 +1,68 @@
+package com.anaplan.engineering.azuki.tictactoe.adapter.scriptgen
+
+import com.anaplan.engineering.azuki.core.system.unsupportedBehavior
+import com.anaplan.engineering.azuki.script.generation.ScriptGenerationCheck
+import com.anaplan.engineering.azuki.script.generation.composeAs
+import com.anaplan.engineering.azuki.tictactoe.adapter.api.*
+import com.anaplan.engineering.azuki.tictactoe.dsl.TicTacToeThen
+
+object TicTacToeScriptGenerationCheckFactory : TicTacToeCheckFactory {
+
+    override val game = GameScriptGenerationCheckFactory
+    override val player = PlayerScriptGenCheckFactory
+}
+
+fun interface TicTacToeScriptGenerationCheck : ScriptGenerationCheck<TicTacToeGenerationEnvironment> {
+
+    override val behavior get() = unsupportedBehavior
+}
+
+object GameScriptGenerationCheckFactory : GameCheckFactory {
+
+    override fun hasPlayOrder(gameName: String, players: List<String>) = TicTacToeScriptGenerationCheck {
+        TicTacToeScriptingHelper.scriptifyFunction(TicTacToeThen::gameHasPlayOrder, gameName, players)
+    }
+
+    override fun hasToken(gameName: String, playerName: String, position: Position) = TicTacToeScriptGenerationCheck {
+        TicTacToeScriptingHelper.scriptifyFunction(TicTacToeThen::boardHasToken, gameName, playerName, position)
+    }.composeAs { e -> e.composeBoardCheck(gameName) { addToken(playerName, position) } }
+
+    override fun hasSpace(gameName: String, position: Position) = TicTacToeScriptGenerationCheck {
+        TicTacToeScriptingHelper.scriptifyFunction(TicTacToeThen::boardHasSpace, gameName, position)
+    }.composeAs { e -> e.composeBoardCheck(gameName) { addSpace(position) } }
```

Comment:
⛏️ The name `addSpace` feels a little jarring to me... a "space" on the board denotes the absence of tokens, so it's not something I can add to a board.